app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,53 @@
+class PostsController < ApplicationController
+
+  before_action :authenticate_user!, :except => [:index, :show]
```

Comment:
Style/HashSyntax: Use the new Ruby 1.9 hash syntax. (https://github.com/bbatsov/ruby-style-guide#hash-literals)